investd/__main__.py

```diff
@@ -5,27 +5,37 @@
 
 import click
 
-from investd import reports
-from investd.config import PERSIST_PATH
-from investd.quotes import generate_quotes_csv
-from investd.sources import ingest_sources_as_df
+from investd import quotes, reports, sources
+from investd.config import INVESTD_PERSIST, init_dirs
+from investd.transaction import TX_FILENAME
 
-app_name = "investd"
-log = logging.getLogger(app_name)
+APP_NAME = "investd"
+log = logging.getLogger(APP_NAME)
 
+HELP_TEXT = f"""{APP_NAME} - A tool for summarizing investments.
 
-cli = click.Group(name=app_name, help="investd - A tool for summarizing investments.")
+How it works: Investd read accounts statements from a INVESTD_SOURCES directory,
+process them and write HTML reports to the INVESTD_REPORTS directory.
+
+By default, both INVESTD_SOURCES and INVESTD_REPORTS directories are located in the
+INVESTD_DATA dir. Each of these locations can be configured by environment variables.
+If INVESTD_DATA environment variable is not provided, a dir named 'investd_data'
+at the current working directory will be created and used as such.
+"""
+
+
+cli = click.Group(name=APP_NAME, help=HELP_TEXT)
 
 
 @cli.command(name="ingest-sources")
 @click.option(
     "--output",
     type=click.Path(dir_okay=False, writable=True),
-    default=PERSIST_PATH / "tx.csv",
+    default=INVESTD_PERSIST / TX_FILENAME,
     help="Output path",
 )
 def ingest_sources_cmd(output: Path):
-    df_tx = ingest_sources_as_df()
+    df_tx = sources.ingest_all()
```

Comment:
for some reason ingest_all gives me an error:
>           raise ValueError("len(arrays) must match len(columns)")
> E           ValueError: len(arrays) must match len(columns)

when I added list() wrapper to de-lazify both columns and data, it works

Reply:
I think it happens when sources are empty

Reply:
Confirmed here, it happens when sources is empty. Added the list comprehensions.